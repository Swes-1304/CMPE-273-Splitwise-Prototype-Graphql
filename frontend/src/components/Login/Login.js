import React, { Component } from 'react';
import '../../App.css';
import axios from 'axios';
import cookie from 'react-cookies';
import { Redirect } from 'react-router';
import Navheader from '../navbar/navbar';
import '../navbar/navbar.css';

// Define a Login Component
class Login extends Component {
  // call the constructor method
  constructor(props) {
    // Call the constrictor of Super class i.e The Component
    super(props);
    // maintain the state required for this component
    this.state = {
      email: '',
      password: '',
    };
    // Bind the handlers to this class
    this.emailChangeHandler = this.emailChangeHandler.bind(this);
    this.passwordChangeHandler = this.passwordChangeHandler.bind(this);
    this.submitLogin = this.submitLogin.bind(this);
  }

  componentWillMount() {
    this.setState({
      redirecttohome: null,
    });
  }

  // username change handler to update state variable with the text entered by the user
  emailChangeHandler = (e) => {
    this.setState({
      email: e.target.value,
    });
  };

  // password change handler to update state variable with the text entered by the user
  passwordChangeHandler = (e) => {
    this.setState({
      password: e.target.value,
    });
  };

  // submit Login handler to send a request to the node backend
  submitLogin = async (e) => {
    // prevent page from refresh
    e.preventDefault();

    const { email, password } = this.state;
    if (email === '') {
      alert('Please enter email address');
      this.setState({
        errorMessage1: 'Please enter email address!',
      });
      return;
    }
    const data = {
      email,
      password,
    };
    this.setState({
      errorMessage1: '',
    });
    // set the with credentials to true
    axios.defaults.withCredentials = true;
    // make a post request with the user data
    axios
      .post('http://localhost:3001/login', data)
      .then((response) => {
        console.log('Status Code : ', response.status);
        console.log('response ', response.data);
        if (response.status === 200) {
          console.log(response.data);
          console.log(response.data.username);
          const resuserid = response.data.user_id;
          const resusername = response.data.username;
          const resemail = response.data.email;
          const resprofpic = response.data.profilepic;
          const rescurrency = response.data.currencydef;
          sessionStorage.setItem('userid', resuserid);
          sessionStorage.setItem('username', resusername);
          sessionStorage.setItem('useremail', resemail);
          sessionStorage.setItem('profilepic', resprofpic);
          sessionStorage.setItem('defaultcurrency', rescurrency);
          const redirectVar1 = <Redirect to="/dashboard" />;
          this.setState({
            redirecttohome: redirectVar1,
          });
        } else {
          console.log(response.data);
          alert(response.data);
          this.setState({
            redirecttohome: null,
          });
        }
      })
      .catch((err) => {
        console.log(err.response.data);
        alert(err.response.data);
        this.setState({
          errorMessage: err.response.data,
        });
      });
  };

  render() {
    let redirectVar = null;
    if (cookie.load('cookie')) {
      redirectVar = <Redirect to="/dashboard" />;
    }
    const { errorMessage, errorMessage1 } = this.state;
    const { redirecttohome } = this.state;
    return (
      <div>
        {redirectVar}
        {redirecttohome}
        <Navheader />
        <div className="container">
          <div className="login-form">
            <div className="main-div">
              <div className="panel">
                <h2>WELCOME TO SPLITWISE!</h2>
              </div>
              <div className="form-group">
                <label htmlFor="email">
                  EMAIL ADDRESS
                  <input
                    type="text"
                    className="form-control"
                    name="email"
                    id="email"
                    placeholder="Email"
                    onChange={this.emailChangeHandler}
                    required
                    formNoValidate
                  />
                </label>
                <p className="errmsg" style={{ color: 'maroon' }}>
                  {' '}
                  {errorMessage1}{' '}
                </p>
              </div>
              <div className="form-group">
                <label htmlFor="email">
                  PASSSWORD
                  <input
                    type="password"
                    className="form-control"
                    name="password"
                    id="password"
                    placeholder="Password"
                    onChange={this.passwordChangeHandler}
                    required
                    formNoValidate
                  />
                </label>
              </div>
              <button
                data-testid="login"
                type="submit"
                className="login-default "
                onClick={this.submitLogin}
                formNoValidate
              >
                Login
              </button>
              <p className="errmsg" style={{ color: 'maroon' }}>
                {' '}
                {errorMessage}{' '}
              </p>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
// export Login Component
export default Login;
