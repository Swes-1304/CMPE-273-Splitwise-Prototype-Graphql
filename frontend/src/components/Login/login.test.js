import React from 'react';
import { render, cleanup } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import Login from './Login';
import Navheader from '../navbar/navbar';

const component = TestRenderer.create(
  <MemoryRouter>
    <Login>
      <Navheader />
    </Login>{' '}
  </MemoryRouter>
);

// eslint-disable-next-line no-undef
afterEach(cleanup);

it('renders', async () => {
  expect(component.toJSON()).toMatchSnapshot();
});

test('Check for login butoon', async () => {
  const { getByTestId } = render(
    <MemoryRouter>
      <Login>
        <Navheader />
      </Login>{' '}
    </MemoryRouter>
  );
  expect(getByTestId('login')).toHaveTextContent('Login');
});
