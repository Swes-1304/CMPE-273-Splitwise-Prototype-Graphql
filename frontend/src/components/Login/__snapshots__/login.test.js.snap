// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Array [
  <div>
    <div>
      <nav
        className="navbar-default navbar navbar-expand navbar-light"
      >
        <a
          className="navbar-brand"
          classname="Navbar-Brand"
          href="/"
          variant="light"
        >
          <img
            alt="icon"
            className="icon"
            src="/Group_Photos/Splitwiseicon.png"
          />
          <span
            className="NavItem"
          >
            Splitwise
          </span>
        </a>
        <div
          aria-expanded={null}
          className="justify-content-end navbar-collapse collapse"
        >
          <div
            className="navbar-nav"
            onKeyDown={[Function]}
          >
            <ul
              className="nav navbar-nav navbar-right"
            >
              <li>
                <button
                  className="login-default btn btn-primary"
                  disabled={false}
                  type="button"
                >
                  <a
                    href="/login"
                    onClick={[Function]}
                  >
                    Login 
                  </a>
                </button>
                or
                 
                <button
                  className="Signup-default btn btn-primary"
                  disabled={false}
                  type="button"
                >
                  <a
                    href="/signup"
                    onClick={[Function]}
                  >
                    Sign Up
                  </a>
                </button>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
    <div
      className="container"
    >
      <div
        className="login-form"
      >
        <div
          className="main-div"
        >
          <div
            className="panel"
          >
            <h2>
              WELCOME TO SPLITWISE!
            </h2>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="email"
            >
              EMAIL ADDRESS
              <input
                className="form-control"
                formNoValidate={true}
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                required={true}
                type="text"
              />
            </label>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="email"
            >
              PASSSWORD
              <input
                className="form-control"
                formNoValidate={true}
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                required={true}
                type="password"
              />
            </label>
          </div>
          <button
            className="login-default "
            data-testid="login"
            formNoValidate={true}
            onClick={[Function]}
            type="submit"
          >
            Login
          </button>
          <p
            className="errmsg"
            style={
              Object {
                "color": "maroon",
              }
            }
          >
             
             
          </p>
        </div>
      </div>
    </div>
  </div>,
  " ",
]
`;
